crates/sui-core/src/gateway_state.rs

```diff
@@ -638,7 +638,11 @@ where
                     pending_transaction,
                 } => {
                     debug!(tx_digest=?pending_transaction, "Objects locked by a previous transaction, re-executing the previous transaction");
-                    if self.retry_pending_tx(pending_transaction).await.is_err() {
+                    if let Err(err) = self.retry_pending_tx(pending_transaction).await {
+                        debug!(
+                            "Retrying pending tx failed: {:?}. Resetting the transaction lock",
+                            err
+                        );
```

Comment:
FYI if you use `tap::TapFallable` you can do this more concisely as:

```
if self.retry_pending_tx(pending_transaction)
    .await
    .tap_err(|e| debug!("Retrying pending tx failed: {:?}. Resetting the transaction lock", err))
    .is_err() {
```

up to you